src/org/oidc/service/util/ServiceUtil.java

```diff
@@ -1,15 +1,56 @@
 package org.oidc.service.util;
 
+import com.auth0.msg.Message;
+import com.google.common.base.Strings;
+import java.net.MalformedURLException;
+import java.net.URL;
+import org.oidc.common.SerializationType;
+import org.oidc.common.UnsupportedContentType;
+
 /**
  * This class has utility methods for various services
  **/
 public class ServiceUtil {
     /**
      Pick out the fragment or query part from a URL.
      * @param url a URL possibly containing a query or a fragment part
-     * @return the query/fragment part
+     * @return the query/reference part
      **/
-    public static String getUrlInfo(String url) {
-        return null;
+    public static String getUrlQueryReference(String url) throws MalformedURLException {
```

Comment:
when are you throwing MalformedURLException?

Reply:
when instantiating new URL(), it throws that exception